service/src/test/java/bio/terra/landingzone/library/landingzones/definition/factories/CromwellBaseResourcesFactoryTest.java

```diff
@@ -38,60 +36,44 @@ void setUp() {
-    var resources =
-        landingZoneManager
-            .deployLandingZoneAsync(
-                landingZoneId,
-                CromwellBaseResourcesFactory.class.getSimpleName(),
-                DefinitionVersion.V1,
-                null)
-            .collectList()
-            .block();
-
-    // Note that this resource list does not include pre-requisite resources
-    assertThat(resources, hasSize(6));
+    landingZoneManager
+        .deployLandingZoneAsync(
+            landingZoneId,
+            CromwellBaseResourcesFactory.class.getSimpleName(),
+            DefinitionVersion.V1,
+            null)
+        .collectList()
+        .block();
 
     // check if you can read lz resources
-    TimeUnit.SECONDS.sleep(3); // wait for tag propagation...
-    var sharedResources = landingZoneManager.reader().listSharedResources(landingZoneId);
-    assertThat(sharedResources, hasSize(7));
+    await() // wait for tag propagation...
+        .atMost(Duration.ofSeconds(120))
+        .untilAsserted(
+            () -> {
+              var landingZoneResources =
+                  landingZoneManager.reader().listAllResources(landingZoneId);
+              assertThat(landingZoneResources, hasSize(9));
+            });
 
+    assertThat(landingZoneManager.reader().listSharedResources(landingZoneId), hasSize(8));
     assertHasVnetWithPurpose(landingZoneId, SubnetResourcePurpose.WORKSPACE_COMPUTE_SUBNET);
     assertHasVnetWithPurpose(landingZoneId, SubnetResourcePurpose.AKS_NODE_POOL_SUBNET);
     assertHasVnetWithPurpose(landingZoneId, SubnetResourcePurpose.WORKSPACE_BATCH_SUBNET);
     assertHasVnetWithPurpose(landingZoneId, SubnetResourcePurpose.POSTGRESQL_SUBNET);
-  }
-
-  @Test
-  void deleteLandingZoneResources_resourcesAreDeleted()
-      throws InterruptedException, LandingZoneRuleDeleteException {
-    String landingZoneId = UUID.randomUUID().toString();
-    var resources =
-        landingZoneManager
-            .deployLandingZoneAsync(
-                landingZoneId,
-                CromwellBaseResourcesFactory.class.getSimpleName(),
-                DefinitionVersion.V1,
-                null)
-            .collectList()
-            .block();
 
-    var deletedResources = landingZoneManager.deleteResources(landingZoneId);
+    landingZoneManager.deleteResources(landingZoneId);
 
     // Immediate listing after deletion may return transient resources results.
     await()
         .atMost(Duration.ofSeconds(120))
-        .until(
+        .untilAsserted(
             () -> {
-              Stream<GenericResource> stream =
-                  armManagers
-                      .azureResourceManager()
-                      .genericResources()
-                      .listByResourceGroup(resourceGroup.name())
-                      .stream();
-              return stream.count() == 0;
+              var landingZoneResources =
+                  landingZoneManager.reader().listAllResources(landingZoneId);
```

Comment:
Switch from listing all resources in resource group to all in landing zone sometimes Azure creates hidden resources which means deleting the landing zone does not necessarily leave the resource group empty. I realize that this misses resource that don't get LZ tags but I think this is more of a fallacy of using tags to track LZ resources. Given we can't be certain resource count will go to 0, I think this is the best we can do.

Reply:
The delete operation should leave the resource group empty. What is the case where resources a left behind?

Reply:
I can't make it happen now but I was seeing some hidden alert resource